### ISSUE TITLE
documenting behaviors when running nu as login shell

### ISSUE THREAD

**Issue by uncomfyhalomacro** [nushell/nushell#10316]
# Documenting Nushell's Behavior as a Login Shell

## Non POSIX Compliant
Obviously, nushell is **NOT** a *POSIX-compliant shell*. This may be the reason why users have trouble using it as a login shell in GNU/Linux systems. These may lead to some weird behaviors such as #934 and #7941. So far these two are the first documented issues I can reproduce. There may be others too so if you have anything you can share here, please chime in.

# Workarounds

## Nushell as a login shell does not inherit /etc/locale.conf 

To explicitly do add the following to `login.nu`

```nu
$env.LANG = "en_US.UTF-8"
$env.LC_ALL = "en_US.UTF-8"
```

## Nushell does not auto update `XDG_DATA_DIRS`

This behavior messes up the ability to set default applications or even launching one through `gtk-launch` e.g. flatpaks, if it's not added in `XDG_DATA_DIRS`. To resolve the issue, one must add the following to `login.nu`:

```nu
$env.XDG_DATA_DIRS = $"/var/lib/flatpak/exports/share:/usr/local/share/:/usr/share/:($env.HOME)/.local/share:($env.HOME)/.local/share/flatpak/exports/share"
```

# Other known issues

> [!WARNING]
> It is *always considered* as bad practice to use `#!/usr/bin/sh` in scripts. One must explicitly  use the full name of the executable e.g. `bash`, `zsh`. Otherwise, setting `nu` as a default login shell may make your POSIX-compliant scripts be launched by `nu`. This is resolved in https://github.com/nushell/nushell/pull/8951 but please be aware for those that may not have that fix.

**Comment by fdncred** [comment]
Just following up from Discord. The intent is to document this in our book and then probably close this issue. Maybe here  https://www.nushell.sh/book/default_shell.html#setting-nu-as-login-shell-linux-bsd-macos and here https://www.nushell.sh/book/configuration.html#configuring-nu-as-a-login-shell.

**Comment by uncomfyhalomacro** [comment]
I will probably wait more until tomorrow. Then I am going to edit the book for the time being. 🫡

**Comment by otakutyrant** [comment]
It is embrassing to find out My nushell is not a POSIX-compliant shell and it cannot automatically inherit some environment variables from /etc/profile.d today.

**Comment by eisterman** [comment]
Hello, seems that when nushell is runned as login shell with `nu -l` manually, my config.nu fails because I use `$env.HOME` inside it and it seems that it is NOT defined during the config.nu run if the shell is in login mode.

Obviously I solved by using an hardcoded path, but I just wanted to report it here.

I think it's because if it's runned in a clean environment, nu doesn't define HOME because as said is not POSIX, so it just use the one inherited from the outer system, that in login or a clean system can be not setted

**Comment by fdncred** [comment]
Yes, if nushell runs as your login shell it does not load any files other than nushell files. So, you must specify what you need manually.

I wouldn't be opposed to changing that for MacOS and Linux but it would need to work with all major distros.
